athena-federation-sdk-tools/src/main/java/com/amazonaws/athena/connector/validation/FederationServiceProvider.java

```diff
@@ -0,0 +1,108 @@
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK Tools
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connector.validation;
+
+import com.amazonaws.athena.connector.lambda.request.PingRequest;
+import com.amazonaws.athena.connector.lambda.request.PingResponse;
+import com.amazonaws.athena.connector.lambda.security.FederatedIdentity;
+import com.amazonaws.athena.connector.lambda.serde.VersionedObjectMapperFactory;
+import com.amazonaws.services.lambda.AWSLambdaClientBuilder;
+import com.amazonaws.services.lambda.invoke.LambdaFunction;
+import com.amazonaws.services.lambda.invoke.LambdaFunctionNameResolver;
+import com.amazonaws.services.lambda.invoke.LambdaInvokerFactory;
+import com.amazonaws.services.lambda.invoke.LambdaInvokerFactoryConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.amazonaws.athena.connector.lambda.handlers.SerDeVersion.SERDE_VERSION;
+import static com.amazonaws.athena.connector.validation.ConnectorValidator.BLOCK_ALLOCATOR;
+
+public class FederationServiceProvider
+{
+    private static final Logger log = LoggerFactory.getLogger(FederationServiceProvider.class);
+
+    private static final String UNKNOWN_SUFFIX = "_unknown";
```

Comment:
I think I'd prefer that we change this to "_validation" or something else that makes it clearer where this request is originating.

Reply:
Changed.  Made LambdaMetadataProvider and LambdaRecordProvider use this method too